deeppavlov/core/agent/rich_content.py

```diff
@@ -97,6 +100,10 @@ class RichMessage(RichItem):
     def __init__(self) -> None:
         self.controls: list = []
 
+    def __str__(self, *args, **kwargs):
+        result = '\n'.join([str(control) for control in self.controls if str(control)])
```

Comment:
If you want to be cool, you can
```suggestion
        result = '\n'.join(filter(bool, map(str, self.controls)))
```